File: week15/week15-3.cpp
int equalSubstring(char* s, char* t, int maxCost) {
    int n=strlen(s);
    int ans=0,j=0;
    for(int i=0;i<n;i++){
        maxCost-=abs(s[i]-t[i]);
        while(maxCost<0){
            maxCost+=abs(s[j]-t[j]);
            j+=1;
        }
        if(i-j+1>ans) ans=i-j+1;
    }
    return ans;
}

File: week15/week15-3.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        n=len(s)
        ans=0
        j=0
        for i in range (n):
            maxCost-=abs(ord(s[i])-ord(t[i]))
            while maxCost<0:
                maxCost+=abs(ord(s[j])-ord(t[j]))
                j+=1
            ans=max(ans,i-j+1)
        return ans